packages/ods/src/components/pagination/src/components/ods-pagination/ods-pagination.tsx

```diff
@@ -0,0 +1,338 @@
+
+            if (isLeft) {
+              this.handlePreviousClick(Number(this.current));
+            } else {
+              this.handleNextClick(Number(this.current));
+            }
+          }}
+          onKeyDown={(event: KeyboardEvent) => {
+            if (isLeft) {
+              this.handlePreviousKeyDown(event, Number(this.current));
+            } else {
+              this.handleNextKeyDown(event, Number(this.current));
+            }
+          }}
+          variant={ODS_BUTTON_VARIANT.ghost}
+          size={ODS_BUTTON_SIZE.md}
+        >
+        </ods-button>
+        {tooltipLabel && (
+          <ods-tooltip
+            shadowDomTriggerId={arrowButtonId}
+            triggerId={this.hostId}
+          >
+            {tooltipLabel}
+          </ods-tooltip>
+        )}
+      </li>
+    );
+  }
+
+  renderEllipsis(): typeof Fragment {
+    return (
+      <li>
+        <ods-button
+          class="ods-pagination__list__page__ellipsis"
+          color={ ODS_BUTTON_COLOR.primary }
+          isDisabled={ true }
+          label="&#x2026;"
+          variant={ ODS_BUTTON_VARIANT.ghost }
+        >
+        </ods-button>
+      </li>
+    );
+  }
+
+  render(): FunctionalComponent | undefined {
+    if (!this.totalItems && this.actualTotalPages < 2) {
+      return;
+    }
+
+    return (
+      <Host
+        class="ods-pagination"
+        isDisabled={this.isDisabled}
+        id={ this.hostId }
+      >
+        {
+          !!this.totalItems &&
+            <div class={{
+              'ods-pagination__results': true,
+              'ods-pagination__results--disabled': this.isDisabled,
+            }}>
+              {
+                this.totalItems >= ODS_PAGINATION_PER_PAGE.min &&
+                  <ods-select isDisabled={this.isDisabled}
+                    value={`${this.itemPerPage}`}
+                    name="ods-pagination__items-per-page"
+                  >
+                    {
+                      ODS_PAGINATION_PER_PAGE_OPTIONS.map((option) => (
+                        <option key={option} value={option}>{option}</option>
+                      ))
+                    }
+                  </ods-select>
+              }
+              <ods-text preset={ ODS_TEXT_PRESET.label }>
+                <slot name="before-total-items"></slot>
+                {this.totalItems}
+                <slot name="after-total-items"></slot>
+              </ods-text>
+            </div>
+        }
+
+        <ul class="ods-pagination__list">
+          { this.renderArrows( 'left' ) }
+
+          {
+            this.pageList
+              .filter((page) => page.active)
+              .map((page) => {
+                const pageId = this.pageList.indexOf(page) + 1;
+
+                this.pageList.length > 6 && this.pageList.length - this.current > 3 && pageId === this.pageList.length && this.renderEllipsis();
+
+                this.pageList.length > 6 && this.current > 4 && pageId === 1 && this.renderEllipsis();
```

Comment:
replace those two inline with a single function that manages both cases
\+ use global constants instead of magic numbers